packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -416,18 +422,41 @@ async function doGenerateArmTemplate(ctx: SolutionContext): Promise<Result<any,
       >;
       if (result.isOk()) {
         bicepOrchestrationTemplate.applyTemplateV2(pluginWithArm.name, result.value);
+        if (result.value.Configuration?.Modules) {
+          for (const module of Object.entries(result.value.Configuration.Modules)) {
+            const moduleName = module[0];
+            const moduleValue = module[1] as string;
+            moduleConfigFiles.set(generateBicepModuleConfigFilePath(moduleName), moduleValue);
+          }
+        }
+
         if (result.value.Provision?.Modules) {
           for (const module of Object.entries(result.value.Provision.Modules)) {
             const moduleName = module[0];
             const moduleValue = module[1] as string;
-            moduleFiles.set(generateBicepModuleProvisionFilePath(moduleName), moduleValue);
+            moduleProvisionFiles.set(generateBicepModuleProvisionFilePath(moduleName), moduleValue);
           }
         }
+      } else {
+        const msg = format(
+          getStrings().solution.GenerateArmTemplateFailNotice,
+          ctx.projectSettings?.appName
+        );
+        ctx.logProvider?.error(msg);
+        return result;
+      }
+    } else if (pluginWithArm.updateArmTemplates) {
+      const pluginContext = getPluginContext(ctx, pluginWithArm.name);
```

Comment:
no need to run update arm templates for selected plugin?

Reply:
sure